packages/backend/routes/devcon.js

```diff
@@ -0,0 +1,94 @@
+const express = require("express");
+const { ethers } = require("ethers");
+const db = require("../services/db/db");
+const { verifySignature } = require("../utils/sign");
+
+const router = express.Router();
+const SRE_BACKEND = "https://scaffold-directory-dev.ew.r.appspot.com";
+
+/**
+ * Check if a builder is eligible for a Devcon voucher based on their status and creation date
+ * Eligibility criteria:
+ * 1. BG batch members created before end of August 2024 (type: batch)
+ * 2. BG member created before end of August 2024 (type: builder)
+ * 3. SRE builders with at least 3 challenges completed and joined before end of August 2024 (type: builder)
+ */
+async function checkEligibility(builderAddress) {
+  const END_OF_AUGUST_2024 = 1725148799000;
+  const builder = await db.findUserByAddress(builderAddress);
+
+  if (builder?.data?.builderBatch && builder?.data?.creationTimestamp < END_OF_AUGUST_2024) {
+    return { isEligible: true, type: "batch" };
+  }
+
+  if (builder.exists && builder?.data?.creationTimestamp < END_OF_AUGUST_2024) {
+    return { isEligible: true, type: "builder" };
+  } else {
```

Comment:
nit: we can remove this `else` since we are doing early return above

Reply:
Good one! https://github.com/scaffold-eth/buidlguidl-v3/pull/113/commits/b03bfdb9371ea70a22541c79803c39678c7393b2